lib/matplotlib/colorbar.py

```diff
@@ -356,35 +425,108 @@ def draw_all(self):
+                elif isinstance(self.norm, colors.SymLogNorm):
+                    # The subs setting here should be replaced
+                    # by logic in the locator.
+                    locator = ticker.SymmetricalLogLocator(
+                                      subs=np.arange(1, 10),
+                                      linthresh=self.norm.linthresh,
+                                      base=10)
+                else:
+                    if mpl.rcParams['_internal.classic_mode']:
+                        locator = ticker.MaxNLocator()
+                    else:
+                        locator = _ColorbarAutoLocator(self)
+            else:
+                b = self._boundaries[self._inside]
+                locator = ticker.FixedLocator(b, nbins=10)
+        _log.debug('locator: %r', locator)
+        return locator, formatter
+
+    def _use_auto_colorbar_locator(self):
+        """
+        Return if we should use an adjustable tick locator or a fixed
+        one.  (check is used twice so factored out here...)
+        """
+        return (self.boundaries is None
+                and self.values is None
+                and ((type(self.norm) == colors.Normalize)
+                    or (type(self.norm) == colors.LogNorm)))
+
     def update_ticks(self):
         """
         Force the update of the ticks and ticklabels. This must be
         called whenever the tick locator and/or tick formatter changes.
         """
         ax = self.ax
-        ticks, ticklabels, offset_string = self._ticker()
-        if self.orientation == 'vertical':
-            ax.yaxis.set_ticks(ticks)
-            ax.set_yticklabels(ticklabels)
-            ax.yaxis.get_major_formatter().set_offset_string(offset_string)
-
+        # get the locator and formatter.  Defaults to
+        # self.locator if not None..
+        locator, formatter = self._get_ticker_locator_formatter()
+
+        if self._use_auto_colorbar_locator():
+            _log.debug('Using auto colorbar locator on colorbar')
+            _log.debug('locator: %r', locator)
+            if self.orientation == 'vertical':
+                long_axis, short_axis = ax.yaxis, ax.xaxis
+            else:
+                long_axis, short_axis = ax.xaxis, ax.yaxis
+            long_axis.set_major_locator(locator)
+            long_axis.set_major_formatter(formatter)
+            if type(self.norm) == colors.LogNorm:
+                long_axis.set_minor_locator(_ColorbarLogLocator(self,
+                            base=10., subs='auto'))
+                long_axis.set_minor_formatter(ticker.NullFormatter())
         else:
-            ax.xaxis.set_ticks(ticks)
-            ax.set_xticklabels(ticklabels)
-            ax.xaxis.get_major_formatter().set_offset_string(offset_string)
+            _log.debug('Using fixed locator on colorbar')
+            ticks, ticklabels, offset_string = self._ticker(locator, formatter)
+            if self.orientation == 'vertical':
```

Comment:
You can use the long_axis, short_axis here, too.  Initialize them prior to the `_use_auto_colorbar_locator` check.